src/codegen/expression.rs

```diff
@@ -389,6 +389,61 @@ pub fn expression(
             *loc,
             Box::new(expression(expr, cfg, contract_no, ns, vartab)),
         ),
+        Expression::DynamicArrayPush(loc, array, ty, value) => {
+            let elem_ty = match ty {
+                Type::Array(..) => match ty.array_elem() {
```

Comment:
Use `array_deref()` and you should not need to do the match.

Reply:
Looks like `array_deref()` doesn't work here, because it always wraps a layer of `Ref` before it gets to the actual array element type.